lib/lan_connection.js

```diff
@@ -130,21 +130,29 @@ Scanner.prototype.start = function() {
       try {
         // Start discovering Tessels
         self.browser.discover();
-      } catch (error) {
-        self.emit('error', error);
+      } catch (err) {
+        process.nextTick(function() {
+          self.emit('error', err);
+        });
       }
     });
 
     // When the browser finds a new device
     self.browser.on('update', function(data) {
-      // Check if it's a Tessel
-      if (_.findIndex(self.discovered, data) > -1) {
-        return;
+      try {
+        // Check if it's a Tessel
+        if (_.findIndex(self.discovered, data) > -1) {
+          return;
+        }
+
+        var connection = new LANConnection(data);
+
+        self.emit('connection', connection);
+      } catch (err) {
+        process.nextTick(function() {
+          self.emit('error', err);
+        });
```

Comment:
The `emit` function will issue a throw right away which will get caught by the faulty library.

[emit function, lib/events.js L70-L92](https://github.com/joyent/node/blob/master/lib/events.js#L70-L92)